crates/agent/src/lib.rs

```diff
@@ -101,6 +102,48 @@ where
     }
 }
 
+#[derive(serde::Deserialize)]
+struct EncryptedHMACKeys {
```

Comment:
nit: HMACKeys? they're not encrypted.
Since this is single-use, I'll suggest inline-ing next to its usage.

But, stepping back, perhaps the `decrypt_hmac_keys` should take a &mut tables::DataPlane?